Generation of gene expression matrix
The raw sequences were transformed to Cell Ranger61 format using a custom python script modified from MISAR-seq23. The resulting fastq files were then aligned to the mouse genome (mm39) and counted using Cell Ranger (v8.0.0), yielding feature-barcode matrices for downstream analysis.
For compatibility with SpotClean25, spot information files were constructed using a custom R script based on the barcode index and the spot coordinates derived from the tissue masks. This data frame contained the following fields: barcode (unique barcode sequence), tissue (0 = not in tissue, 1 = in tissue), imagerow and imagecol (indices of the Y and X barcodes, respectively), and height and width (no practical significance). The raw featureâ€“barcode matrix was subsequently decontaminated using SpotClean, resulting in a final expression matrix with spatial locations as rows and gene expression levels as columns. Spatial heatmaps for pan-mRNA or individual genes were generated using the ggplot2 R package.
# HiMA-seq
